src/Lean/Server/FileWorker.lean

```diff
@@ -310,6 +310,16 @@ section Updates
           validSnaps := validSnaps.dropLast
       unfoldCmdSnaps newMeta validSnaps.toArray cancelTk ctx
     modify fun st => { st with doc := ⟨newMeta, AsyncList.delayed newSnaps, cancelTk⟩ }
+
+  /-- called when document is closed so we can clear any diagnostics and cancel pending work -/
+  def exitWorker : WorkerM Unit := do
```

Comment:
Nit: I would put this right at the beginning of the `MainLoop` section as it is not an update handler.

Reply:
Done.